packages/pigeon/lib/java_generator.dart

```diff
@@ -0,0 +1,145 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'ast.dart';
+import 'generator_tools.dart';
+
+/// Options that control how Java code will be generated.
+class JavaOptions {
+  /// The name of the class that will house all the generated classes.
+  String className;
+
+  /// The package where the generated class will live.
+  String package;
+}
+
+void _writeHostApi(Indent indent, Api api) {
+  assert(api.location == ApiLocation.host);
+  indent.writeln('/** Generated interface from pigeon. */');
+  indent.write('public interface ${api.name} ');
+  indent.scoped('{', '}', () {
+    for (Method method in api.methods) {
+      indent.writeln(
+          '${method.returnType} ${method.name}(${method.argType} arg);');
+    }
+    indent.addln('');
+    indent.writeln(
+        '/** Sets up an instance of `${api.name}` to handle messages through the `binaryMessenger` */');
```

Comment:
Nit, since the Java class api.name is something you made, it'll still not be clear what an instance of it does specifically. Mention that the instance is meant to hold a message handler and that it's needed for the java side to receive signals from flutter or some such.

Reply:
Expanded on the docstrings for the interfaces and generated classes.